### PR TITLE
Better documentation for nuisance_model and heterogeneity_model

### PR DESCRIPTION
**Pull request by gaugup**
## Description

It seems there was little insight on which nuisance_model/heterogeneity_model is to be used. 

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [x] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [x] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: responsibleai/responsibleai/_managers/causal_manager.py [hunk 1]
```diff
@@ -92,6 +92,13 @@ def add(
         :param heterogeneity_features: Features that mediate the causal effect.
         :type heterogeneity_features: list
         :param nuisance_model: Model type to use for nuisance estimation.
```
Comment (romanlutz) [inline responsibleai/responsibleai/_managers/causal_manager.py:94]: There's some redundancy there between l.94 and 96. Perhaps the following is better:

```
:param nuisance_model: Model type to use for nuisance estimation.
            (i.e. for estimating propensity models or models of how controls predict the outcome)
            It is one of {'linear', 'automl'}, optional (default='linear').
            If 'linear', then LassoCV (for regression) and
            LogisticRegressionCV (for classification) are used.
            If 'automl', then a k-fold cross-validation and model selection
            is performed among several models and the best is chosen.
```

**Review by romanlutz — APPROVED** [review]

File: responsibleai/responsibleai/_managers/causal_manager.py (hunk 1 above)
Comment (riedgar-ms) [reply responsibleai/responsibleai/_managers/causal_manager.py:94]: The line 'estimating propensity models or models of how controls predict the outcome' still isn't terribly clear to me.... I don't understand what this is really doing.

Comment (gaugup) [reply responsibleai/responsibleai/_managers/causal_manager.py]: I took this doc from econml. For now I have removed this bit and only added description for linear and automl models.

File: responsibleai/responsibleai/managers/causal_manager.py [hunk 2]
```diff
@@ -91,7 +91,12 @@ def add(
         :type treatment_features: list
         :param heterogeneity_features: Features that mediate the causal effect.
         :type heterogeneity_features: list
-        :param nuisance_model: Model type to use for nuisance estimation.
+        :param nuisance_model: Model type to use for nuisance estimation .
+            It is one of {'linear', 'automl'}, optional (default='linear').
```
Comment (riedgar-ms) [inline responsibleai/responsibleai/managers/causal_manager.py:95]: Could you put in a sentence as to what 'nuisance' means for a causal model? I know that this isn't the place to put an in-depth discussion, but this only means something to someone who already knows about the subject area.

File: responsibleai/responsibleai/_managers/causal_manager.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/causal_manager.py]: @riedgar-ms, I have updated the description for nuisance_model and hetrogenity_model. Can you take a look and approve?

File: responsibleai/responsibleai/managers/causal_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/causal_manager.py:96]: @riedgar-ms, I have updated the description for nuisance_model and hetrogenity_model. Can you take a look and approve?

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**